src/libasr/pass/array_op.cpp

```diff
@@ -182,6 +182,38 @@ class ArrayOpVisitor : public PassUtils::PassVisitor<ArrayOpVisitor>
         }
     }
 
+    void visit_Module(const ASR::Module_t &x) {
+        if (strcmp(x.m_name, "_global_symbols") == 0) {
+            // FIXME: this is a hack, we need to pass in a non-const `x`,
```

Comment:
try removing this if check. If it does not work, let's figure out the real bug why it doesn't work.